File: apps/codemap-search/tests/fixtures/navigation/cpp/controller.cpp
#include "controller.hpp"

class CheckoutController {
public:
    Response post(const Request &raw) {
        auto request = parser_.parse(raw);
        auto receipt = service_.submit(request);
        return Response::ok(receipt);
    }

private:
    OrderParser parser_;
    CheckoutService service_;
};

File: apps/codemap-search/tests/fixtures/navigation/javascript/controller.js
import { Response } from "./http";
import { parseOrder } from "./parser";

export class CheckoutController {
  constructor(service) {
    this.service = service;
  }

  async post(request) {
    const payload = parseOrder(request.body);
    const result = await this.service.submit(payload);
    return Response.ok(result);
  }
}
